tests/test_engine.py

```diff
@@ -433,6 +433,10 @@ def test_close_downloader(self):
         e = ExecutionEngine(get_crawler(MySpider), lambda _: None)
         yield e.close()
 
+        if hasattr(e, "downloader"):
```

Comment:
I think this needs to be a separate test method. And I think it should be possible to replicate an early failure (e.g. by passing a `DOWNLOADER` value that cannot be used to create an instance) instead of deleting e.downloader directly.